diff --git a/src/cmd/go/build.go b/src/cmd/go/build.go
index 3c0b994..e591c40 100644
--- a/src/cmd/go/build.go
+++ b/src/cmd/go/build.go
@@ -358,7 +358,8 @@ func buildModeInit() {
 		ldBuildmode = "c-shared"
 	case "default":
 		switch platform {
-		case "android/arm", "android/arm64", "android/amd64", "android/386":
+		case "linux/386", "linux/amd64", "linux/arm", "linux/arm64", "linux/ppc64le",
+			"android/arm", "android/arm64", "android/amd64", "android/386":
 			codegenArg = "-shared"
 			ldBuildmode = "pie"
 		case "darwin/arm", "darwin/arm64":
